gdal/frmts/fits/fitsdataset.cpp

```diff
@@ -599,6 +734,689 @@ GDALDataset *FITSDataset::Create( const char* pszFilename,
+                    oSRS.SetProjParm(SRS_PP_LATITUDE_OF_ORIGIN, crval2);
+                    oSRS.SetProjParm(SRS_PP_LATITUDE_OF_ORIGIN, crval2);
+
+                /* Lambert Azimuthal Equal Area / ZEA projection */
+                } else if( strcmp(pstr,"ZEA" ) == 0 ) {
+                    projName.assign("Lambert_Azimuthal_Equal_Area_");
+                    oSRS.SetProjection(SRS_PT_LAMBERT_AZIMUTHAL_EQUAL_AREA);
+                    oSRS.SetProjParm(SRS_PP_LONGITUDE_OF_CENTER, crval1);
+                    oSRS.SetProjParm(SRS_PP_LATITUDE_OF_CENTER, crval2);
+
+                /* Lambert Conformal Conic 1SP / COO projection */
+                } else if( strcmp(pstr,"COO" ) == 0 ) {
+                    projName.assign("Lambert_Conformal_Conic_1SP_");
+                    oSRS.SetProjection(SRS_PT_LAMBERT_CONFORMAL_CONIC_1SP);
+                    oSRS.SetProjParm(SRS_PP_LONGITUDE_OF_CENTER, crval1);
+                    /*
+                    #clat = self.__header['XXXXX']
+                    #srs.SetProjParm('latitude_of_center',clat)
+                    scale = self.__header['XXXXX']
+                    if scale is not None:
+                        srs.SetProjParm('scale_factor',scale)
+                    else: #set default of 1.0
+                        srs.SetProjParm('scale_factor',1.0)
+                    */
+
+                /* Orthographic / SIN projection */
+                } else if( strcmp(pstr,"SIN" ) == 0 ) {
+                    projName.assign("Orthographic_");
+                    oSRS.SetProjection(SRS_PT_ORTHOGRAPHIC);
+                    oSRS.SetProjParm(SRS_PP_CENTRAL_MERIDIAN, crval1);
+                    /*
+                    #olat = self.__header['XXXXX']
+                    #srs.SetProjParm('latitude_of_origin',olat)
+                    */
+
+                /* Point Perspective / AZP projection */
+                } else if( strcmp(pstr,"AZP" ) == 0 ) {
+                    projName.assign("perspective_point_height_");
+                    oSRS.SetProjection(SRS_PP_PERSPECTIVE_POINT_HEIGHT);
+                    /* # appears to need height... maybe center lon/lat */
+
+                /* Polar Stereographic / STG projection */
+                } else if( strcmp(pstr,"STG" ) == 0 ) {
+                    projName.assign("Polar_Stereographic_");
+                    oSRS.SetProjection(SRS_PT_POLAR_STEREOGRAPHIC);
+                    oSRS.SetProjParm(SRS_PP_CENTRAL_MERIDIAN, crval1);
+                    oSRS.SetProjParm(SRS_PP_LATITUDE_OF_ORIGIN, crval2);
+                } else {
+                    CPLError(CE_Failure, CPLE_AppDefined,
+                         "Unknown projection.");
+                }
+
+                projName.append(target);
+                oSRS.SetProjParm(SRS_PP_FALSE_EASTING,0.0);
+                oSRS.SetProjParm(SRS_PP_FALSE_NORTHING,0.0);
+
+                char * cstrproj = new char [projName.length()+1];
```

Comment:
useless memory allocation, and memory leak.
Just do `oSRS.SetNode("PROJCS",projName.c_str());`
Same for cstrgeoname and cstrdname below

Reply:
Fixed in https://github.com/OSGeo/gdal/pull/1298/commits/202323471002d33a3acfcc0549fe8312d9e2d680